vm/devices/storage/disk_nvme/nvme_driver/src/namespace.rs

```diff
@@ -499,6 +520,70 @@ impl Namespace {
 
         Ok(())
     }
+
+    /// Return Namespace ID.
+    pub fn nsid(&self) -> u32 {
+        self.nsid
+    }
+
+    /// Save namespace object data for servicing.
+    pub fn save(&self) -> anyhow::Result<SavedNamespaceData> {
```

Comment:
Why is this not using the vmcore::SaveRestore trait?

Reply:
For simple objects like this one I think I can make the change. There might be complexities with top-level objects such as nvme worker, or more broadly, with the objects which require async.

Reply:
We discussed this offline - the trait doesn't fit into well for drivers where we need to create the object first. This would mean it would be in some sort of uninitialized state and need to support that, vs a default new path. So its fair enough not to use the trait meant for devices.

We owe some more documentation on this.